[17.0][MIG] repair_picking

JasminSForgeFlow opened the pull request:
Standard Migration

Move from https://github.com/OCA/manufacture/tree/14.0/repair_picking

@ForgeFlow

JoanSForgeFlow committed:
[ADD] repair_picking

LoisRForgeFlow committed:
[FIX] repair_picking: readme structure + issue in tests

Commit:
[UPD] Update repair_picking.pot

OCA-git-bot committed:
[BOT] post-merge updates

DavidJForgeFlow committed:
[IMP] repair_picking: add dependency to repair_stock, and moving there common methods.

Commit:
[UPD] Update repair_picking.pot

OCA-git-bot committed:
[BOT] post-merge updates

DavidJForgeFlow committed:
[REF] repair_picking: move stock_rule._get_custom_move_fields to repair_stock_move

mymage committed:
Added translation using Weblate (Italian)

JasminSForgeFlow committed:
[IMP] repair_picking: black, isort, prettier

JasminSForgeFlow committed:
[MIG] repair_picking: Migration to 17.0

DavidJForgeFlow commented on repair_picking/models/repair.py:
```diff
@@ -0,0 +1,167 @@
+            "repair_line_id": line.id,
+            "repair_id": line.repair_id.id,
+        }
+        # if line.lot_id:
```
Why is this commented? It would not work if you don't change it to:
```
if line.lot_id:
    procurement_data["restrict_lot_id"] = line.lot_id.id
```

LoisRForgeFlow replied:
Fixed!

LoisRForgeFlow replied:
It is not that easy, because now the line is a stock.move and does not have a fixed lot_id. I will propose something

LoisRForgeFlow replied:
I think we should depend on this https://github.com/OCA/repair/pull/88 to allow to select the lot in the repair lines.

LoisRForgeFlow committed:
[FIX] repair_picking: add depencency to repair_restrict_lot

this allow to specify the lot/serial in the repair lines and
keep the functionality based on that in this module.

LoisRForgeFlow commented:
Testing the approach in last commit. New dependency:

* https://github.com/OCA/repair/pull/88

LoisRForgeFlow committed:
[FIX] repair_picking: fix migration

* Adapt to new repair line type "recycle".
* Propoerly integrate with the other modules in the repo (like `repair_stock` and `repair_type`).
* Simplify logic and make it a bit more robust.
* Add more test lines to make tests more useful.

LoisRForgeFlow commented:
@DavidJForgeFlow could you check this one?

DavidJForgeFlow reviewed (approved):
Seems to be working fine now! Code also LGTM!

LoisRForgeFlow commented:
/ocabot merge nobump
